File: backend/src/main/java/com/campussync/campussync_backend/dto/UpdateStudentRequest.java
package com.campussync.campussync_backend.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class UpdateStudentRequest {

    @NotBlank
    private String name;

    private String phoneNumber;

    @NotBlank
    private String programme;

    @NotNull
    private Integer admissionYear;

    @NotNull
    private Long departmentId;

    @NotNull
    private Long branchId;

    @NotNull
    private Integer semester;

    @NotNull
    private Integer graduationYear;

    private boolean internal;
}

File: frontend/lib/models/update_student_request.dart
class UpdateStudentRequest {
  final String name;
  final String? phoneNumber;
  final String programme;
  final int admissionYear;
  final int departmentId;
  final int branchId;
  final int semester;
  final int graduationYear;
  final bool internal;

  UpdateStudentRequest({
    required this.name,
    this.phoneNumber,
    required this.programme,
    required this.admissionYear,
    required this.departmentId,
    required this.branchId,
    required this.semester,
    required this.graduationYear,
    required this.internal,
  });

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'phoneNumber': phoneNumber,
      'programme': programme,
      'admissionYear': admissionYear,
      'departmentId': departmentId,
      'branchId': branchId,
      'semester': semester,
      'graduationYear': graduationYear,
      'internal': internal,
    };
  }
}